File: contest/leetcode/hamming-distance.cpp
/* coding:utf-8
 * Copyright (C) dirlt
 */

class Solution {
   public:
    int hammingDistance(int x, int y) {
        int z = x ^ y;
        int dist = 0;
        while (z) {
            dist += (z & 0x1);
            z = z >> 1;
        }
        return dist;
    }
};

File: leetcode/hamming-distance.py
#!/usr/bin/env python
# coding:utf-8
# Copyright (C) dirlt

class Solution:
    def hammingDistance(self, x: int, y: int) -> int:
        z = x ^ y
        ans = 0
        for i in range(32):
            if z & (1 << i):
                ans += 1
        return ans
